src/enzyme_ad/jax/Passes/LibDeviceFuncsRaisingPass.cpp

```diff
@@ -500,6 +500,70 @@ struct ReadOnlyAllocaElim : public OpRewritePattern<LLVM::AllocaOp> {
   }
 };
 
+struct SelectExtractElementToExtractElementSelect
```

Comment:
can we put this in a common utility function so we don't duplicate it in both libdevice and canonicalize-scf-for (though perhaps it becomes no longer necessary there)

Reply:
Done. Note that the libdevice operates on LLVM selects whereas the canonicalize-for operates on arith selects.

Reply:
do we need an llvm select one? the libdevice pattern itself should create arith selects so the arith select one should still apply?